tests/receivers/mongodb/mongodb_discovery_test.go

```diff
@@ -24,18 +24,49 @@ import (
 	"runtime"
 	"syscall"
 	"testing"
+	"time"
 
 	"github.com/docker/docker/api/types/container"
 	"github.com/docker/docker/api/types/network"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
 	"github.com/testcontainers/testcontainers-go"
-	"github.com/testcontainers/testcontainers-go/wait"
+	"go.opentelemetry.io/collector/component/componenttest"
+	"go.opentelemetry.io/collector/consumer/consumertest"
+	"go.opentelemetry.io/collector/pdata/plog"
+	"go.opentelemetry.io/collector/receiver/otlpreceiver"
+	"go.opentelemetry.io/collector/receiver/receivertest"
 )
 
 type otelContainer struct {
 	testcontainers.Container
 }
 
-func mongoDBAutoDiscoveryHelper(ctx context.Context, configFile string, logMessageToAssert string) (*otelContainer, error) {
+const (
+	ReceiverTypeAttr         = "discovery.receiver.type"
+	MessageAttr              = "discovery.message"
+	OtelEntityAttributesAttr = "otel.entity.attributes"
```

Comment:
Can't we use constants from `internal/common/discovery`?

Reply:
@dmitryax This is the same reason, we are trying to avoid not using TestUtils. When we bring this module underneath tests/go.mod it starts a circular dependency issues. I propose eventually, when we get rid of testutils  all together we can use it or may be move this somewhere inside tests package to reuse.

Reply:
Ok. I believe we should move testutils from tests to `internal` even before it can be removed. It should not be exposed outside anyway. Then, we can have tests->internal dependencies

Reply:
Not needed in this PR. Maybe as a follow up if you have time

Reply:
I will do it after this gets merged when I removed TestDockerObsever.